android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java

```diff
@@ -321,6 +312,25 @@ private void deliverEventToGestureHandler(GestureHandler handler, MotionEvent ev
     }
   }
 
+  /**
+   * isViewAttachedUnderWrapper checks whether all of parents for view related to handler
+   * view are attached. Since there might be an issue rarely observed when view
+   * has been detached and handler's state hasn't been change to canceled, failed or
+   * ended yet. Probably it's a result of some race condition and stopping delivering
+   * for this handler and changing its state to failed of end appear to be good enough solution.
+   */
+  private boolean isViewAttachedUnderWrapper(View view) {
+    if (view == mWrapperView) {
+      return true;
+    }
+
+    if (view == null || !(view.getParent() instanceof View)) {
+      return false;
+    }
+
+    return isViewAttachedUnderWrapper((View) view.getParent());
```

Comment:
you changed simple loop to arraycopy few line earlier and here you do a recursion instead of a simple while loop?